modules/components/analytics/core/kpi/org.wso2.carbon.pc.analytics.core.kpi/src/main/java/org/wso2/carbon/pc/analytics/core/kpi/internal/PCAnalticsServiceComponent.java

```diff
@@ -57,32 +55,24 @@ protected void setProcessCenter(ProcessCenterService processCenterService) {
                 getInstance().setStreamAdminServiceClient(new StreamAdminServiceClient(analyticsUrl));
                 getInstance().setReceiverAdminServiceClient(new ReceiverAdminServiceClient(analyticsUrl));
             } catch (Throwable e) {
-                log.error("Failed to set Process Center servcie..", e);
+                log.error("Failed to set Process Center service..", e);
             }
         }
     }
 
-    protected void unsetProcessCenter(
-            ProcessCenterService processCenterService) {
-        if (log.isDebugEnabled()) {
-            log.debug("ProcessCenter unbound from the Process Center KPI Analytics Publisher component");
-        }
+    protected void unsetProcessCenter(ProcessCenterService processCenterService) {
+        log.debug("ProcessCenter unbound from the Process Center KPI Analytics Publisher component");
         getInstance().setProcessCenter(null);
     }
 
     protected void setRegistryService(RegistryService registrySvc) {
-        if (log.isDebugEnabled()) {
```

Comment:
As I think, using if check is more efficient only if there is a string concatenate operation (or some other costly operation) in the argument passed to debug method call. Here, as there is only a simple single String literal passed to the debug method, if check could reduce efficiency. WDYT ?